src/Solver/Interoperability.cpp

```diff
@@ -705,11 +728,11 @@ void seissol::Interoperability::initializeFault( char*   modelFileName,
     seissol::initializers::FaultGPGenerator queryGen( seissol::SeisSol::main.meshReader(),
                                                       reinterpret_cast<double(*)[2]>(bndPoints),
                                                       numberOfBndPoints );
-    parameterDB.evaluateModel(std::string(modelFileName), queryGen);
+    parameterDB.evaluateModel(std::string(modelFileName), &queryGen);
   } else {
     seissol::initializers::FaultBarycentreGenerator queryGen( seissol::SeisSol::main.meshReader(),
                                                               numberOfBndPoints );
-    parameterDB.evaluateModel(std::string(modelFileName), queryGen);
+    parameterDB.evaluateModel(std::string(modelFileName), &queryGen);
```

Comment:
delete queryGen at the end of the function?

Reply:
This is a stack variable.